File: backend/features/crime-map/routes/crimeMapRoutes.js
const express = require("express");
const router = express.Router();
const { authenticate } = require("../../../shared/middleware/tokenMiddleware");
const {
  getBoundaries,
  getPins,
  getStatistics,
  getHeatmap,           // ← new
} = require("../controllers/crimeMapController");

router.get("/boundaries",  authenticate, getBoundaries);
router.get("/pins",        authenticate, getPins);
router.get("/statistics",  authenticate, getStatistics);
router.get("/heatmap",     authenticate, getHeatmap);   // ← new

module.exports = router;

File: backend/features/crime_map/crime_map_routes.py
from fastapi import APIRouter, Depends
from features.crime_map.crime_map_controller import (
    get_boundaries,
    get_pins,
    get_statistics,
    get_heatmap,
)
from shared.middleware.token_middleware import authenticate

router = APIRouter()

@router.get("/boundaries")
async def boundaries(user=Depends(authenticate)):
    return await get_boundaries(user)

@router.get("/pins")
async def pins(user=Depends(authenticate)):
    return await get_pins(user)

@router.get("/statistics")
async def statistics(user=Depends(authenticate)):
    return await get_statistics(user)

@router.get("/heatmap")
async def heatmap(user=Depends(authenticate)):
    return await get_heatmap(user)